lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -335,17 +342,16 @@ follow nl tr cps yield rollback header =
             liftIO $ logInfo tr $ mconcat
                 [ "Applying blocks [", pretty slFst, " ... ", pretty slLst, "]" ]
 
-            yield blocks nodeTip >>= handle cursor' "Failed to roll forward: "
+            yield blocks nodeTip >>= handle cursor'
 
-        Right (RollBackward cursor') -> do
-            let point = cursorSlotId nl cursor'
-            logInfo tr $ "Rolling back to " <> pretty point
-            rollback point >>= handle cursor' "Failed to roll backward: "
```

Comment:
Alternatively to this PR, I guess we could have made `DB.rollback` and the backward-callback fail when a matching checkpoint don't exist.